nipype/algorithms/misc.py

```diff
@@ -850,22 +853,26 @@ def _run_interface(self, runtime):
         try:
             import pandas as pd
         except ImportError:
-            raise ImportError('This interface requires pandas (http://pandas.pydata.org/) to run.')
+            raise ImportError(('This interface requires pandas '
+                               '(http://pandas.pydata.org/) to run.'))
 
         try:
             import lockfile as pl
             self._have_lock = True
         except ImportError:
-            import warnings
-            warnings.warn(('Python module lockfile was not found: AddCSVRow will not be thread-safe '
-                          'in multi-processor execution'))
+            from warnings import warn
+            warn(('Python module lockfile was not found: AddCSVRow will not be'
+                  ' thread-safe in multi-processor execution'))
 
         input_dict = {}
         for key, val in self.inputs._outputs.items():
             # expand lists to several columns
+            if key == 'trait_added' and val in self.inputs._outputs.keys():
```

Comment:
Hi @satra, I'm not very happy with this way of checking that ```'trait_added'``` is not interpreted as one more output. Is there a way to check this more cleanly?. If not, this PR should be merged as it fixes the bad behavior in case there are Undefined inputs.